client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -0,0 +1,133 @@
+import { useColorPaletteVariations, useFontPairingVariations } from '@automattic/global-styles';
+import { keyBy } from '@automattic/js-utils';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { ONBOARD_STORE } from '../../../../../stores';
+import type { Pattern, Category } from '../types';
+import type { OnboardSelect } from '@automattic/data-stores';
+import type { Design } from '@automattic/design-picker/src/types';
+import type { GlobalStylesObject } from '@automattic/global-styles';
+
+const useRecipe = ( siteId = 0, patterns: Pattern[], categories: Category[] ) => {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

Can I ask what `recipe` mean in this PR? Is there something called `recipe` about onboarding?

Reply:
Maybe PCYsg-HDF-p2. It's just a pre-defined page layout.

Reply:
Thank you!
I've looked a bit at how `state.selectedDesign (recipe)` is used, and I see that we use it to hold the selected design during onboarding (in my understanding).